doc/lsst.cp.pipe/constructing-calibrations.rst

```diff
@@ -0,0 +1,733 @@
+
+.. _cp-pipe-certification
+
+
+Verification of calibrations
+============================
+
+The goal of the `cp_verify` package is to provide a largely automated way to determine the quality of newly produced calibrations.  This is done by applying the newly constructed calibration to a set of test exposures, measuring some set of metrics on the resulting images, and comparing those metrics to the tests documented in DMTN-101.  It should be run as part of a general inspection of the calibration prior to use.  However, as this is still in development, and hasn't been extended to all calibration types, it is still only a recommended step.  Manually examining the output calibration (or its effect on a few test exposures) is still highly recommended before large scale processing is performed.
+
+
+Calibration certification
+=========================
+
+Certification of calibrations should be done after confirming that all of the ``cp_verify`` tests have passed, and any tests that still fail should be documented on the ticket managing the construction.  Calibrations created as part of daily observations should be certified into a single collection per calibration type, with the validity range set to only include the day they were taken, such as:
+.. code:: bash
+
+    butler certify-calibrations /repo/main \
+        u/czw/DM-XYZ/biasGen.20210715 LATISS/calib/dailyBiases \
+        --begin_date 2021-07-15 --end_date 2021-07-16 bias
+
+Longer term master calibrations should be certified into ticketed CALIBRATION collection (containing however many calibration types and date ranges that entails), that should then be chained to the instrument's recommended CALIBRATION collection.  The results of ``cp_verify`` should provide some guidance on the validity range, as additional raw exposures can be checked against the proposed calibration to identify when the verification tests fail.  The end date may be best set to a future date that will allow the calibration to be used until a new one supersedes it:
+.. code:: bash
+
+    butler certify-calibrations /repo/main \
+        u/czw/DM-XYZ/biasGen.20210715 LATISS/calib/DM-XYZ \
+        --begin_date 2021-01-01 --end_date 2050-01-01 bias
+
+Calibration collection best practices
+-------------------------------------
+
+Although the example presented below certifies each new calibration to a final calibration collection, in situations where there is some question at to the quality of the calibration, it may be better to use a chained collection as the target.  Fixing an error in the example presented a way to demonstrate this as well.
+
+- The initial bias and defects were correct, and a chained collection was used:
+.. code:: bash
+
+   butler certify-calibrations /repo/main \
```

Comment:
I thought it would be less confusing to not use variables, as they'd be single use in most cases.

Reply:
If this is general documentation rather than "how to use NCSA butler repository" then by convention you could always refer to any butler repository as `$BUTLER_REPO` and not give an explicit path ever.